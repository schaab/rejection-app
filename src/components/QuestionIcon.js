import React from "react";
import PropTypes from "prop-types";
import AcceptedIcon from "./AcceptedIcon";
import RejectedIcon from "./RejectedIcon";

const REJECTED = "Rejected";

const QuestionIcon = ({ status = REJECTED }) =>
  status === REJECTED ? <RejectedIcon /> : <AcceptedIcon />;

QuestionIcon.displayName = "QuestionIcon";
QuestionIcon.propTypes = {
  status: PropTypes.string
};

export default QuestionIcon;
