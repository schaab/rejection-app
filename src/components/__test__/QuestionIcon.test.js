import React from 'react'
import { render, cleanup } from 'react-testing-library'
import 'jest-dom/extend-expect'
import QuestionIcon from '../QuestionIcon'

describe('<QuestionIcon />', () => {
  afterEach(cleanup)

  test('given no props it should render a default component', () => {
    const { container } = render(<QuestionIcon />)
    const actual = container.firstChild

    expect(actual).toMatchSnapshot()
  })

  test('given a rejected status it should render a rejected icon', () => {
    const { container } = render(<QuestionIcon status="Rejected" />)
    const actual = container.firstChild

    expect(actual).toMatchSnapshot()
  })

  test('given an accepted status it should render an accepted icon', () => {
    const { container } = render(<QuestionIcon status="Accepted" />)
    const actual = container.firstChild

    expect(actual).toMatchSnapshot()
  })
})
