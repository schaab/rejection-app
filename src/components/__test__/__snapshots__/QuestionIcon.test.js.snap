// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionIcon /> given a rejected status it should render a rejected icon 1`] = `
<div
  class="MuiAvatar-root-13 MuiAvatar-colorDefault-14"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root-16"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M16 8v8H8V8h8m2-2H6v12h12V6z"
    />
  </svg>
</div>
`;

exports[`<QuestionIcon /> given an accepted status it should render an accepted icon 1`] = `
<div
  class="MuiAvatar-root-25 MuiAvatar-colorDefault-26"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root-28"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <g>
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
      <path
        d="M16.59 7.58L10 14.17l-2.59-2.58L6 13l4 4 8-8z"
      />
    </g>
  </svg>
</div>
`;

exports[`<QuestionIcon /> given no props it should render a default component 1`] = `
<div
  class="MuiAvatar-root-1 MuiAvatar-colorDefault-2"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root-4"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M16 8v8H8V8h8m2-2H6v12h12V6z"
    />
  </svg>
</div>
`;
